### PR TITLE
change: add csv deserializer

### PR DESCRIPTION
**Pull request by laurenyu**
*Description of changes:*
a counterpart to our CSV serializer

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#commit-message-guidlines)
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have updated any necessary [documentation](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/predictor.py [hunk 1]
```diff
@@ -229,6 +229,20 @@ def _row_to_csv(obj):
     return ','.join(obj)
 
 
+class _CsvDeserializer(object):
+    def __init__(self):
+        self.content_type = CONTENT_TYPE_CSV
+
+    def __call__(self, stream, content_type):
+        try:
+            return list(csv.reader(stream.read().decode('utf-8').splitlines()))
```
Comment (icywang86rui) [inline src/sagemaker/predictor.py]: should we make encoding configurable as well here?

File: tests/unit/test_predictor.py [hunk 2]
```diff
@@ -141,6 +142,16 @@ def test_csv_serializer_csv_reader():
...
+def test_csv_deserializer_array():
+    result = csv_deserializer(io.BytesIO(b'1,2,3'), 'text/csv')
+    assert result == [['1', '2', '3']]
+
+
+def test_csv_deserializer_2dimensional():
+    result = csv_deserializer(io.BytesIO(b'1,2,3\n3,4,5'), 'text/csv')
+    assert result == [['1', '2', '3'], ['3', '4', '5']]
+
+
```
Comment (icywang86rui) [inline tests/unit/test_predictor.py]: I would add a test with just one line and one value. YMMV

File: src/sagemaker/predictor.py (hunk 1 above)
Comment (laurenyu) [reply src/sagemaker/predictor.py]: good point

File: tests/unit/test_predictor.py (hunk 2 above)
Comment (laurenyu) [reply tests/unit/test_predictor.py]: done

File: src/sagemaker/predictor.py [hunk 3]
```diff
@@ -230,12 +230,13 @@ def _row_to_csv(obj):
 
 
 class _CsvDeserializer(object):
-    def __init__(self):
-        self.content_type = CONTENT_TYPE_CSV
+    def __init__(self, encoding='utf-8'):
```
Comment (icywang86rui) [inline src/sagemaker/predictor.py]: do you need to pass the accept in as well?

Comment (laurenyu) [reply src/sagemaker/predictor.py]: I was staying consistent with:

```python
class _CsvSerializer(object):
    def __init__(self):
        self.content_type = CONTENT_TYPE_CSV
```

open to changing it though if there's a compelling use case

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=98a39cf3-1991-4cfd-bc1c-225c6099b2ef%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=fa817d27-3f42-4d48-9bfb-71d7bc5b5a7e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**